presto-main/src/main/java/com/facebook/presto/sql/planner/ExpressionInterpreter.java

```diff
@@ -332,8 +335,44 @@ else if (javaType == Slice.class) {
         @Override
         protected Object visitDereferenceExpression(DereferenceExpression node, Object context)
         {
-            // Dereference is never a Symbol
-            return node;
+            Type type = expressionTypes.get(node.getBase());
+            // if there is no type for the base of Dereference, it must be QualifiedName
+            if (type == null) {
+                return node;
+            }
```

Comment:
I think it would be impossible since in your example, `t` is a table, if the `node.getBase()` is a table, the `ExpressionAnalyzer` will catch this and there is no corresponding type for `node.getBase()`, then the `type` will be `null`, and the original `node` will be directly returned.